Automattic/newspack-ads
# Rename to newspack-ads

The use case of this plugin has been expanded to cover ad services besides Google Ad Manager, so the plugin name has been genericized from `newspack-gam` to `newspack-ads`. This PR applies this change everywhere, in class names, function names, descriptions, etc. The block has been renamed to `Ad Unit` (slug: `ad-unit`). 

Previously when the block was named `newspack-gam/google-ad-manager` it was possible to add it to the block category "Newspack" because it had been established earlier in the load process by the plugin `newspack-blocks`. Now that this plugin's name comes first, it was necessary to replicate the block category code. This explains the changes to `webpack.config.js` and the `src/setup` directory. 

Companion PR for `newspack-plugin` is here: https://github.com/Automattic/newspack-plugin/compare/update/rename-newspack-ads?expand=1

#### Testing

1) Because the plugin has been renamed, it probably makes sense to delete the old repo and start clean.
2) In `wp-content/plugins/` of a test environment, clone this repo.
3) Checkout `update/rename-to-ads`
4) Run `npm install && npm run build:webpack`
5) `newspack-plugin` needs to be on `update/rename-newspack-ads` (and run `npm run build:webpack` there too)
6) All ad units created before these changes should exist as they were in the Newspack plugin UI.
7) Verify that they can be created, edited, and deleted just as before.
8) The block name has changed, so any previous instances of the block will be invalid and broken.
9) In a post, add new instances of `Ad Unit` block. Verify that these behave as they did previously.
10 Publish or preview the post and verify the ad behaves as before, with AMP and non-AMP codes.
11) Verify all tests pass.
